NarutoLutaDeNinjas-Map/src/main/java/br/tec/db/NinjaDeNinjutsu.java

```diff
@@ -0,0 +1,36 @@
+package br.tec.db;
+
+public class NinjaDeNinjutsu extends Personagem implements Ninja {
+
+    public NinjaDeNinjutsu(String nome, int vida) {
+        super(nome, vida);
+    }
+
+    @Override
+    public void usarJutsu(String jutsuKeyMap, Personagem personagemAtacado) {
```

Comment:
Poderia refatorar algumas sequencia de ações e testes dentro do método em métodos menores. Ajuda na clareza, manutenibilidade e testes.
Por exemplo: 
this.getJutsus().containsKey(jutsuKeyMap) poderia ser um método chamado possuiJutsu() ,
a sequencia para atacar poderia ser um metodo como realizarAtaqueUsandoNinjutsu(... , ...) ,
e a sequencia que verifica se ataque foi efetivo ou não outro metodo também.

Recomendo a utilização do plugin Code Metrics no IntelliJ. Ajuda a manter uma referencia de como está ficando a complexidade do código.